### ISSUE TITLE
Win7 32 bit and 64 bit Enigma Portable: Unable to load library 'compileEGMf'

### ISSUE THREAD

**Issue by whyameye** [enigma-dev/lgmplugin#17]
Downloading 7zip executable from here:
http://enigma-dev.org/docs/Wiki/Install:Windows
Download: https://www.dropbox.com/s/nlag2evgiiis476/ENIGMA%20Portable.exe
Size: 75.64 MB
Last Updated: 5/10/2014
GCC Version: 4.8.2

On a first run I get:
java.lang.UnsatisfiedLinkError: Unable to load library 'compileEGMf': Native library (win32-x86/compileEGMf.dll) not found in resource path ([file:/C:/Users/John/Downloads/ENIGMA/enigma-dev/plugins/enigma.jar])

Tried on 2 different Win7 machines, one 32 bit and one 64 bit.

java -version on 64 bit machine returns:
java version "1.6.0_26"
Java(TM) SE Runtime Environment (build 1.6.0_26-b03)
Java HotSpot(TM) 64-Bit Server VM (build 20.1-b02, mixed mode)

java -version on 32 bit machine returns:
java version "1.7.0_55"
Java(TM) SE Runtime Environment (build 1.7.0_55-b14)
Java HotSpot(TM) Client VM (build 24.55-b03, mixed mode, sharing)

I was testing Enigma for a computer lab of Windows machines at a middle school so building from source seems impractical.

**Comment by pac4jpsnet** [comment]
I did everything above, Had to change the config to "C:/Program Files/ENIGMA/" in the ey file as ProgramData would have merged or replaced my existing Program Files and couldn't risk that.  Still got the same error which I will list below.  Since I have a dual boot of Ubuntu and Win10 and LatGM works flawlessly in Ubuntu, what is the easiest GUI friendly cross compiler to make the Ubuntu created games export to windows.  Haven't yet found a pre-installed MinGw64 for Ubuntu but hoping.
Anyway here is the copy of the error report.  Thanks for any help.


Stack trace:

java.lang.UnsatisfiedLinkError: Unable to load library 'compileEGMf': Native library (win32-x86-64/compileEGMf.dll) not found in resource path ([file:/C:/Users/Carol%20Powell/Desktop/EnigmLatWinV/ENIGMA/enigma-dev/plugins/enigma.jar])
	at com.sun.jna.NativeLibrary.loadLibrary(NativeLibrary.java:271)
	at com.sun.jna.NativeLibrary.getInstance(NativeLibrary.java:398)
	at com.sun.jna.Library$Handler.<init>(Library.java:147)
	at com.sun.jna.Native.loadLibrary(Native.java:412)
	at com.sun.jna.Native.loadLibrary(Native.java:391)
	at org.enigma.EnigmaRunner.attemptLib(EnigmaRunner.java:248)
	at org.enigma.EnigmaRunner.access$1(EnigmaRunner.java:241)
	at org.enigma.EnigmaRunner$3.run(EnigmaRunner.java:179)

**Comment by pac4jpsnet** [comment]
Just discovered this was closed in May 2014. Sorry I missed it. Any redirects to a current similar post?
Thanks.

**Comment by RobertBColton** [comment]
Did you install a 32 bit Java? The ZIP currently only works with a 32 bit Java.
http://enigma-dev.org/docs/Wiki/Install:Parallel_Java

**Comment by pac4jpsnet** [comment]
I don't recall installing Java separately as thought it was in the zip file. I have a 64 bit system of Win10.

Thanks for any ideas.

Carol

On November 18, 2016, at 12:26 AM, Robert B Colton <notifications@github.com> wrote:

Did you install a 32 bit Java? The ZIP currently only works with a 32 bit Java.
http://enigma-dev.org/docs/Wiki/Install:Parallel_Java

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or mute the thread.￼

{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/enigma-dev/lgmplugin","title":"enigma-dev/lgmplugin","subtitle":"GitHub repository","main_image_url":"<a href="https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png">https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png</a>","avatar_image_url":"<a href="https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png">https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png</a>","action":{"name":"Open in GitHub","url":"<a href="https://github.com/enigma-dev/lgmplugin">https://github.com/enigma-dev/lgmplugin</a>"}},"updates":{"snippets":[{"icon":"PERSON","message":"@RobertBColton in #17: Did you install a 32 bit Java? The ZIP currently only works with a 32 bit Java.\r\n<a href="http://enigma-dev.org/docs/Wiki/Install:Parallel_Java">http://enigma-dev.org/docs/Wiki/Install:Parallel_Java</a>"}],"action":{"name":"View Issue","url":"<a href="https://github.com/enigma-dev/lgmplugin/issues/17#issuecomment-261475192">https://github.com/enigma-dev/lgmplugin/issues/17#issuecomment-261475192</a>"}}}

**Comment by pac4jpsnet** [comment]
I have repeated requests for my password reset with an promise of an email
to reset my password, but none have shown up.
I wished to share a major breakthrough with LateralGM: I  have
inadvertently improved it,not only stopping Java crashes but even enabled
it to play the games  immediately using the green button instead of waiting
for the compiled game.   Nonetheless, the information that helped is found
at http://1pixme.com/lateralGMfix.html  for your review.
Hope this helps someone.

Thanks.
Sincerely,
Carol Powell

**Comment by JoshDreamland** [comment]
I don't understand, Carol; did you try getting the Windows install zip to work on Linux? There is a separate set of instructions for installing on Linux....

Also, are you having trouble resetting your ENIGMA forum password? I can help you with that.

**Comment by pac4jpsnet** [comment]
I resolved it by copying older files of Enigma to the new install. It now will play a game when just hitting the green arrow.
Still can't reset my password. Never comes to my inbox.

Thanks for any help you can offer.

Carol

On April 19, 2017, at 6:04 AM, Josh Ventura <notifications@github.com> wrote:

I don't understand, Carol; did you try getting the Windows install zip to work on Linux? There is a separate set of instructions for installing on Linux....

Also, are you having trouble resetting your ENIGMA forum password? I can help you with that.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or mute the thread.￼

{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/enigma-dev/lgmplugin","title":"enigma-dev/lgmplugin","subtitle":"GitHub repository","main_image_url":"<a href="https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png">https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png</a>","avatar_image_url":"<a href="https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png">https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png</a>","action":{"name":"Open in GitHub","url":"<a href="https://github.com/enigma-dev/lgmplugin">https://github.com/enigma-dev/lgmplugin</a>"}},"updates":{"snippets":[{"icon":"PERSON","message":"@JoshDreamland in #17: I don't understand, Carol; did you try getting the Windows install zip to work on Linux? There is a separate set of instructions for installing on Linux....\r\n\r\nAlso, are you having trouble resetting your ENIGMA forum password? I can help you with that."}],"action":{"name":"View Issue","url":"<a href="https://github.com/enigma-dev/lgmplugin/issues/17#issuecomment-295262591">https://github.com/enigma-dev/lgmplugin/issues/17#issuecomment-295262591</a>"}}}

**Comment by bonezinger** [comment]
First off, I'd like to say that I was also getting the compiler error, where the buttons are greyed out, unclickable. Second, I'd also like to note that uninstalling Java 64 bit and installing a 32bit version of did the trick. I believe it'd be nice if this solution got posted/stickied up in this problem, since I doubt i'll be the last person to get it.


Here's the full error I was getting, which is the same as pac4jp was having.
OS: Windows 10
Java 8 64bit version

The exact error message is:
Unable to load library 'compileEGMf': Native library (win-32-x86-64/compileEGMf.dll) not found in resource path ([file:/D:/ENIGMA/enigma-dev/plugins/enigma.jar])

This is the unhandled exception, which is almost the same thing:
```
Operating System: Windows 10
Version: 10.0
Architecture: amd64

Java Name: Java HotSpot(TM) 64-Bit Server VM
Java Vendor: Oracle Corporation
Version: 1.8.0_131

Available processors (cores): 4
Free memory (bytes): 203151368
Maximum memory (bytes): 932184064
Total memory available to JVM (bytes): 257425408

File system root: C:\
Total space (bytes): 195964182528
Free space (bytes): 88745738240
Usable space (bytes): 88745738240

File system root: D:\
Total space (bytes): 280513998848
Free space (bytes): 69891641344
Usable space (bytes): 69891641344

File system root: E:\
Total space (bytes): 0
Free space (bytes): 0
Usable space (bytes): 0

Stack trace:

java.lang.UnsatisfiedLinkError: Unable to load library 'compileEGMf': Native library (win32-x86-64/compileEGMf.dll) not found in resource path ([file:/D:/ENIGMA/enigma-dev/plugins/enigma.jar])
	at com.sun.jna.NativeLibrary.loadLibrary(NativeLibrary.java:271)
	at com.sun.jna.NativeLibrary.getInstance(NativeLibrary.java:398)
	at com.sun.jna.Library$Handler.<init>(Library.java:147)
	at com.sun.jna.Native.loadLibrary(Native.java:412)
	at com.sun.jna.Native.loadLibrary(Native.java:391)
	at org.enigma.EnigmaRunner.attemptLib(EnigmaRunner.java:248)
	at org.enigma.EnigmaRunner.access$1(EnigmaRunner.java:241)
	at org.enigma.EnigmaRunner$3.run(EnigmaRunner.java:179)

```

**The fix: Install JRE Windows x86 (32 bit) instead.**
